app/src/org/commcare/views/media/MediaLayout.java

```diff
@@ -166,10 +166,10 @@ public void onClick(View v) {
                     }
 
                     Intent i = new Intent("android.intent.action.VIEW");
-                    i.setDataAndType(Uri.fromFile(videoFile), "video/*");
+                    Uri videoFileUri = FileUtil.getUriForExternalFile(getContext(), videoFile);
+                    i.setDataAndType(videoFileUri, "video/*");
+                    i.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
                     try {
-                        String uri = Uri.fromFile(videoFile).getPath().replaceAll("^.*\\/", "");
```

Comment:
the only thing it was being used for was the log statement immediately after, which seemed very useless, so I removed them both.